index.js

```diff
@@ -42,13 +42,28 @@ exports.handler = (event, context, callback) => {
       logger.error(`Unrecognized input record: ${JSON.stringify(r)}`);
     });
 
+    // failsafe timeout on pingbacks
+    let failsafe;
+    if (process.env.PINGBACKS) {
+      failsafe = setTimeout(function() {
+        logger.error('Failsafe global timeout!');
+        console.log(JSON.stringify(records));
+        if (process.env.EXIT_ON_PINGBACKS_TIMEOUT) {
+          callback(null, `Failsafe exit`);
+          process.exit(0);
```

Comment:
This _might_ clear the event loop and allow the lambda to exit cleanly, moving on to the next batch of kinesis.  Or it might not.  That's why it's an ENV!